core/bin/oauth.rs

```diff
@@ -203,6 +204,74 @@ async fn connections_access_token(
     }
 }
 
+#[derive(Deserialize)]
+struct CredentialPayload {
+    provider: CredentialProvider,
+    raw_json: String,
```

Comment:
I don't think this should mention "json" at all. IMO it's just a string, the user of this API may decide to use this to store JSON strings, but I don't think this should be requirement (you could store private keys or other non-JSON secrets)

Reply:
I see your point, will change.

We are very close to turn this into a key/value store :)
Why not just use a key/value store btw ?

Reply:
It really is an encrypted key value store i think. We could use a premade one, but imo cost of adding another piece of infra and code to interface with it is higher than adding a couple of endpoints to oauth which we already use for other connectors. Wdyt?

Reply:
It's an interesting question.

I think it's more extensible to store a JSON here no?

Will make a friendlier API

Reply:
The other way around no? If you just store a string, you can store json but also store non-json stuff

Reply:
We should do the same as other “secret” managers do IMO (k8s or gcp just store a string)

Reply:
I'm neutral

Reply:
One argument in favor of enforcing json objects is that it removes the trisk of storing strings that have multiple levels of json.stringify by mistake

Reply:
Yes that's a good point.

I propose we do:
```
#[derive(Deserialize)]
struct CredentialPayload {
    provider: CredentialProvider,
    credentials: serde_json::Value,
}

Reply:
Still need a way to enforce it is a JSON object if we’re going for this (since any string is a serde value)